overlord/configstate/config/transaction.go

```diff
@@ -178,8 +224,17 @@ func (t *Transaction) Get(snapName, key string, result interface{}) error {
 		return err
 	}
 
-	// commit changes onto a copy of pristine configuration, so that get has a complete view of the config.
+	// merge virtual config and then commit changes onto a copy of pristine configuration, so that get has a complete view of the config.
 	config := t.copyPristine(snapName)
+	// 1. merge virtual
+	mergedConfig, err := mergeConfigWithVirtual(snapName, jsonRaw(config))
```

Comment:
it's anyway weird doing the conversions here though, I think mergeConfigWithVirtual should take and return the arguments as available and nedeed by its caller and do the conversions inside itself?

Reply:
Indeed, I changed this now.